model/openapi.yml

```diff
@@ -526,6 +533,64 @@ components:
         fleet:
           description: Agent configuration to describe how to connect to fleet-server.
           type: object
+    actionUnenroll:
+      description: The UNENROLL action data.
+      # unenroll actions have no `data` attribute
+    actionRequestDiagnostics:
+      description: The REQUEST_DIAGNOSTICS action data.
+      # diagnostics actions have no `data` attribute
+    actionPolicyReassign:
+      description: The POLICY_REASSIGN action data.
+      type: object
+      required:
+        - policy_id
+      properties:
+        policy_id:
+          desription: The policy ID the agent has been reassigned to.
+          type: string
+    actionPolicyChange:
+      description: The POLICY_CHANGE action data.
+      type: object
+      required:
+        - policy
+      properties:
+        policy:
+          $ref:  "#/components/schemas/policyData"
+    actionUpgrade:
+      description: the UPGRADE action data.
+      type: object
+      required:
+        - version
+      properties:
+        version:
+          description: The version number that the agent should upgrade to.
+          type: string
+        source_uri:
+          description: The source of the upgrade artifact.
+          type: string
+    actionSettings:
+      description: The SETTINGS action data.
+      type: object
+      properties:
+        log_level:
+          type: string
+          enum:
+            - debug
+            - info
+            - warning
+            - error
+    actionCancel:
+      description: The CANCEL action data.
+      type: object
+      required:
+        - target_id
+      properties:
+        target_id:
+          decription: The action to attempt to cancel.
+          type: string
+    actionInputAction:
+      description: The INPUT_ACTION action data.
+      type: object # FIXME: needs security team to define fields as the action is passed from the agent to their componenets.
```

Comment:
Right now the data attribute for an `INPUT_ACTION` is a generic `map[string]interface{}`. Do we want a more in depth definition?

cc: @aleksmaus @scunningham